### DISCUSSION TITLE
GenericJsxEditor help!

### DISCUSSION THREAD

**Discussion by skullicious** [mdx-editor/editor#629] [General]
Hi guys, wondering if you can point me in the right direction.

I've got a really simple use case to start with, I'm trying to port over my existing MDX-JS functionality to MDXEditor as the inline editing is brilliant.

I've gone though the jsx docs at https://mdxeditor.dev/editor/docs/jsx as well and pulled some of the github repo's but I'm unable to get to where I need to be and I'm not sure what I'm missing.

I'm trying to use the GenericJsxEditor in conjunction with some very simple JSX components (for example a branded header with a text/children input).

It seems to be that I am not able to import the element to then render it (I receive no error and the editor is visible (a generic element renders in it's place). 

![Screenshot 2024-11-10 235724](https://github.com/user-attachments/assets/ac035e72-3af4-47c8-9336-09c97fe45425)

![Screenshot 2024-11-10 231324](https://github.com/user-attachments/assets/675eba33-639d-4d9f-bf74-d68b57a7281e)

![Screenshot 2024-11-10 231522](https://github.com/user-attachments/assets/5426b83c-0165-42fd-85a8-a0d3af070e54)

![Screenshot 2024-11-10 231724](https://github.com/user-attachments/assets/a9877f92-3056-4ab7-a1e8-e3afd4bdf6c1)

![Screenshot 2024-11-10 231653](https://github.com/user-attachments/assets/216ba54b-a03d-4cee-b412-f1cc81b2a9c7)

Thanks!

**Comment by terrinens** [comment]
The flow I understood when I opened the source code and checked is as follows.

We found that the GenericJsxEditor failed to get ReactNode, identified only the variables needed for text conversion, and then converted it to text when running ref.current.getMarkdown().

That's why we had to modify the source code to handle the actual node rendering.

The following link is how I solved it.

https://github.com/mdx-editor/editor/discussions/627#discussioncomment-11189870
